pkg/framework/plugins/removefailedpods/failedpods.go

```diff
@@ -148,6 +148,17 @@ func validateCanEvict(pod *v1.Pod, failedPodArgs *RemoveFailedPodsArgs) error {
 		}
 	}
 
+	if len(failedPodArgs.ExitCodes) > 0 && pod.Status.Phase == v1.PodFailed {
```

Comment:
```suggestion
	if len(failedPodArgs.ExitCodes) > 0
```

listing pods already filters by PodFailed status here:
https://github.com/kubernetes-sigs/descheduler/blob/b95380641fcdb31dee6026d7e9e96f226dbbd424/pkg/framework/plugins/removefailedpods/failedpods.go#L70